src/Datagrid/ListMapper.php

```diff
@@ -163,6 +166,11 @@ public function add($name, $type = null, array $fieldDescriptionOptions = [])
         if (!isset($fieldDescriptionOptions['role']) || $this->admin->isGranted($fieldDescriptionOptions['role'])) {
             // add the field with the FormBuilder
             $this->builder->addField($this->list, $type, $fieldDescription, $this->admin);
+
+            // Ensure batch and action pseudo-fields are tagged as virtual
+            if (\in_array($fieldDescription->getType(), ['actions', 'batch', 'select'], true)) {
```

Comment:
You can use the TYPE_ACTIONS const.

And maybe creating TYPE_BATCH and TYPE_SELECT const can be a nice idea